### PR TITLE
Implemented gameInProgress() under Commons entity, and tests

### LINKED ISSUE #20: FEATURE: As an admin, I can control dates for game play - gameInProgress()
https://github.com/ucsb-cs156-f23/proj-happycows-f23-5pm-2/issues/11

### PR DESCRIPTION
**Pull request by alexkxu**
## Overview
In this PR, issue #20 is addressed. A preliminary implementation of gameInProgress() under the Commons entity is implemented alongside tests covering true and false outputs. The function returns true if today's date is >= start date, and <= last day.

Note that this PR relies on a parameter implemented in #18, and of writing, is under development by @amgadhawari. Temporarily, all I have done regarding this is include a `private LocalDateTime lastDate` variable in the Commons class. Due to this, the changes in this PR are obviously subject to change if merge conflicts arise with #18.

Also, do note that "Acceptance Criteria for Future Issues" under the linked issue #11 are not addressed in this PR.

## Feedback Request (Optional)
Please give suggestions on the following:
- Using LocalDateTime libraries might not be optimal in terms of efficiency when comparing dates (used in four statements), but is a relatively safe option. Could be room for optimization.

## Future Possibilities (Optional)

Some possible points could be:
- This is the enables the "Acceptance Criteria for Future Issues" under #11. These include: add logic to the backends of various jobs that operate on all commons (MilkTheCows, UpdateCowHealth, InstructorReports) where it iterates over the commons. Skip over commons where gameInProgress() returns false. 

## Validation (Optional)
1. Download this branch.
2. Run the project.
3. Go to this page.
4. Test this function.

## Tests
- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues
Closes #20
Linked to #11

### CODE REVIEW

**Comment by alexkxu** [comment]
never mind, my updates to the dependencies nuked a test in the starter code, will continue to debug.

**Comment by alexkxu** [comment]
not sure what's causing the problem here, all Jacoco coverage checks are being met locally.

**Review by hancheng-li — DISMISSED** [review]
Other than the commented out code, LGTM

File: src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java [hunk 1]
```diff
@@ -47,5 +47,16 @@ public class Commons {
     @JsonIgnore
     private List<UserCommons> joinedUsers;
 
+    // returns true if today's date is >= start date, and <= last day
+    public boolean gameInProgress() {
+        LocalDateTime todayDateTime = LocalDateTime.now();
+
+        // boolean output = ( (todayDateTime.isAfter(startingDate) || todayDateTime.equals(startingDate)) 
+        //             && (todayDateTime.isBefore(lastDate) || todayDateTime.equals(lastDate)) );
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java:55]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: src/test/java/edu/ucsb/cs156/happiercows/entities/CommonsTests.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package edu.ucsb.cs156.happiercows.entities;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+// import org.mockito.MockedStatic;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/entities/CommonsTests.java:6]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: src/test/java/edu/ucsb/cs156/happiercows/entities/CommonsTests.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    // @Test
+    // void test_gameInProgress_True_Equals_Last_Date() throws Exception {
+    //     String todayDateTime = LocalDateTime.now().toString();
+    //     LocalDateTime ldt = LocalDateTime.parse(todayDateTime);
+    //     try (MockedStatic<LocalDateTime> mockedLocalDateTime = Mockito.mockStatic(LocalDateTime.class)) {
+    //         mockedLocalDateTime.when(LocalDateTime::now).thenReturn(ldt);
+    //         assertEquals(true, Commons.builder().startingDate(dt2).lastDate(ldt).build().gameInProgress());
+    //     }
+    // }
+}
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/entities/CommonsTests.java:52]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by pconrad — CHANGES_REQUESTED** [review]
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java [hunk 4]
```diff
@@ -28,6 +27,7 @@ public class Commons {
     private double milkPrice;
     private double startingBalance;
     private LocalDateTime startingDate;
+    private LocalDateTime lastDate; // subject to change, see issue #18, linked to #11
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/Commons.java:30]: this comment won't make any sense in the final production repo, so let's remove this and make it as a comment on the PR instead.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**